scripts/linux/buildEdgelet.sh

```diff
@@ -0,0 +1,216 @@
+        ARCH="arm64v8"
+    else
+        echo "Unsupported architecture"
+        exit 1
+    fi
+}
+
+###############################################################################
+# Print usage information pertaining to this script and exit
+###############################################################################
+usage()
+{
+    echo "$SCRIPT_NAME [options]"
+    echo "Note: Depending on the options you might have to run this as root or sudo."
+    echo ""
+    echo "options"
+    echo " -i, --image-name     Image name (e.g. edge-agent)"
+    echo " -P, --project        Project to build image for (e.g. iotedged)"
+    echo " -t, --target-arch    Target architecture (default: uname -m)"
+    echo " -n, --namespace      Docker namespace (default: $DEFAULT_DOCKER_NAMESPACE)"
+    echo " -c, --configuration  Build configuration (default: release)"
+    echo "--bin-dir             Directory containing the output binaries. Either use this option or set env variable BUILD_BINARIESDIRECTORY"
+    exit 1;
+}
+
+print_help_and_exit()
+{
+    echo "Run $SCRIPT_NAME --help for more information."
+    exit 1
+}
+
+print_args()
+{
+    echo "Project:      $EDGELET_DIR/$PROJECT"
+    echo "Arch:         $ARCH"
+    echo "Toolchain:    $TOOLCHAIN"
+    echo "Image:        $DOCKER_IMAGENAME"
+    echo "Namespace:    $DOCKER_NAMESPACE"
+    echo "Dockerfile:   $DOCKERFILE"
+    echo
+}
+
+###############################################################################
+# Obtain and validate the options supported by this script
+###############################################################################
+process_args()
+{
+    save_next_arg=0
+    for arg in "$@"
+    do
+        if [[ ${save_next_arg} -eq 1 ]]; then
+            ARCH="$arg"
+            check_arch
+            save_next_arg=0
+        elif [[ ${save_next_arg} -eq 2 ]]; then
+            PROJECT="$arg"
+            save_next_arg=0
+        elif [[ ${save_next_arg} -eq 3 ]]; then
+            DOCKER_IMAGENAME="$arg"
+            save_next_arg=0
+        elif [[ ${save_next_arg} -eq 4 ]]; then
+            DOCKER_NAMESPACE="$arg"
+            save_next_arg=0
+        elif [[ ${save_next_arg} -eq 5 ]]; then
+            BUILD_CONFIGURATION="$arg"
+            save_next_arg=0
+        elif [[ ${save_next_arg} -eq 6 ]]; then
+            BUILD_BINARIESDIRECTORY="$arg"
+            save_next_arg=0
+        else
+            case "$arg" in
+                "-h" | "--help" ) usage;;
+                "-t" | "--target-arch" ) save_next_arg=1;;
+                "-P" | "--project" ) save_next_arg=2;;
+                "-i" | "--image-name" ) save_next_arg=3;;
+                "-n" | "--namespace" ) save_next_arg=4;;
+                "-c" | "--configuration" ) save_next_arg=5;;
+                "--bin-dir" ) save_next_arg=6;;
+                * ) usage;;
+            esac
+        fi
+    done
+
+    if [[ ${PROJECT,,} == "iotedged" ]]; then
+        LIBC="glibc"
```

Comment:
`iotedged` wants `glibc` because of `libiothsm`?

Reply:
Yes, I did my best to make it work with musl, but gave up eventually

Reply:
Got it.